src/pages/home/report/ReportActionsView.js

```diff
@@ -397,6 +425,68 @@ class ReportActionsView extends React.Component {
         }
     }
 
+    /**
+     * Scrolls to a specific report action ID
+     */
+    scrollToReportActionID() {
+        this.actionIndexID = _.findIndex(this.sortedReportActions, (
+            ({action}) => parseInt(action.reportActionID, 10) === this.props.reportActionID
+        ));
+
+        if (this.actionIndexID !== -1) {
+            ReportScrollManager.scrollToIndex({index: this.actionIndexID, viewPosition: 0.5});
+            this.setState({selectedReportActionID: this.props.reportActionID});
+        }
+    }
+
+    /**
+     * Records when a report actionID is done rendering.
+     *
+     * @param {string} reportActionID
+     */
+    recordReportActionIDRendered(reportActionID) {
+        this.renderedActionIDs.add(parseInt(reportActionID, 10));
+        this.checkScrollToReportAction();
+    }
+
+    /**
+     * Records when our FlatList is done measuring the heights and offset of items. Since native will hit this block immediately we'll use a param checkScrollToReportAction to prevent
+     * calling the scroll method unnecessarily until the item has been rendered.
+     *
+     * @param {boolean} checkScrollToReportAction
+     */
+    recordMeasurementDone(checkScrollToReportAction = true) {
+        this.doneMeasuring = true;
+        if (checkScrollToReportAction) {
+            this.checkScrollToReportAction();
+        }
+    }
+
+    /**
+     * Determine if we can scroll now or not.
+     * On web we can scroll only after measurements of all rendered items are done while on native we can scroll after rendering of the specific item is done.
+     */
+    checkScrollToReportAction() {
+        if (!this.props.reportActionID) {
+            return;
+        }
+
+        if (this.doneMeasuring && !this.doneScrollingToReportActionID) {
+            if (this.renderedActionIDs.has(this.props.reportActionID)) {
+                this.doneScrollingToReportActionID = true;
+
+                // We give a slight delay because if we attempt this too fast the scroll gets buggy on iOS as more items need to render.
+                setTimeout(this.scrollToReportActionID, 500);
+            } else if (this.renderedActionIDs.size === this.sortedReportActions.length && this.scrollToReportActionIDAttempt < 3) {
+                this.loadMoreChats();
+                ++this.scrollToReportActionIDAttempt;
```

Comment:
What is the `scrollToReportActionIDAttempt` stuff doing?

Reply:
So this variable is so that we attempt loading more chats and scroll back 3 times. We have this because if you have currently loaded 50 chats but want to visit the 120th oldest chat, this block will load chats (50-100) -> not find the reportActionID -> Load more chats (100-150) -> find the reportActionID and scroll to it. I wanted to place a limit on how far back you can scroll and not allow an infinite loadChats to go 500 messages back etc and hence use this variable for it.

Reply:
Ohhhh, Ok I was not getting that at all. It's because we don't have bidirectional pagination? So instead we try to scroll back a few times to load the chat. 

I have an alternate suggestion that I think could be a little cleaner...

- When the chat loads we look at the max sequence number for the chat and also whether we have the report action in the array or not.
- If we don't have it we look to see how far back we'd have to go
- If it's very far just don't link to the chat
- Otherwise, call fetchActions() and request everything up to the report action we need (or maybe a bit beyond that)
- And then we can scroll when the new data arrives.

Reply:
So firstly we don't have the reportActionID in the array right, thats why we call loadMoreChats.
But then how do you suggest we look to see how far back we'd have to go? Say we've loaded 50 reportActionIDs which is from 5111 to 4550 from newest to oldest. Now the reportActionIDs are rowID so 50 reportActionIDs for one reportID are not immediately sequential (like its not 5,4,3,2,1 but it could be 5,3,1) so if we want to scroll to reportActionID 3660 we do not know how far back it is from 4550, it could be the 55th chat or the 155th chat. Hence we wouldn't know until we load more chats. Anyway i'll reinvestigate if i'm mistaken but if you feel im missing something too let me know 👍

Reply:
>So firstly we don't have the reportActionID in the array right

Well, I am just saying that when the chat mounts it pulls `reportActions` from storage and _might_ have the `reportActionID`. 

>But then how do you suggest we look to see how far back we'd have to go? Say we've loaded 50 reportActionIDs which is from 5111 to 4550 from newest to oldest. Now the reportActionIDs are rowID so 50 reportActionIDs for one reportID are not immediately sequential (like its not 5,4,3,2,1 but it could be 5,3,1) so if we want to scroll to reportActionID 3660 we do not know how far back it is from 4550, it could be the 55th chat or the 155th chat.

Sorry, I'm not following the stuff about rowID or non-sequential or why it matters.

Mainly I was trying to say we can fetch the actions with a formula like this

limit = `report.maxSequenceNumber` - `reportActionID` + some buffer n (for historical actions to fill things out a bit)

Reply:
I am assuming an action with id 3660 would be `(4550 - 3660)` spaces away from 4550. So maybe someone can explain to me why that is not true first 😄

Reply:
oh hah no thats not true. reportActionID is the rowID and hence there can be different reportID values in between. For example, take this dataset,
```
sqlite> select rowID, * from reportActions order by created desc limit 10;
rowid  created              reportID  accountID  action      message
-----  -------------------  --------  ---------  ----------  ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
7027   2022-06-01 22:25:24  224       1          ADDCOMMENT  {"html":"test 1100","clientID":"1654122323882408"}
7026   2022-06-01 22:09:57  700       1          ADDCOMMENT  {"html":"Recent message. comment 1 - <a href=\"http:\/\/localhost:8080\/r\/215\/5923\" target=\"_blank\" rel=\"noreferrer noopener\">http:\/\/localhost:8080\/r\/215\/5923<\/a>","clientID":"1654121396853857"}
7025   2022-06-01 22:09:53  700       1          ADDCOMMENT  {"html":"Recent message. comment 1 - http:\/\/localhost:8080\/r\/215\/5923","clientID":"1654121392754386"}
7024   2022-06-01 22:09:18  700       1          ADDCOMMENT  {"html":"Recent message. comment 1 - http:\/\/localhost:8080\/r\/215\/5923","clientID":"1654121358062389"}
7023   2022-06-01 18:36:53  224       1          ADDCOMMENT  {"html":"test 1099","clientID":"1650325317315275"}
7022   2022-06-01 18:36:53  224       1          ADDCOMMENT  {"clientID":"1650325317315275","edits":["test 1098"],"html":""}
```

For reportID 224 the first comment shown (sequence number 1) is reportActionID 7027, its next actionID is not 7027-1 spaces and is instead 7023 because there are other reportID comment actions in the middle.

Reply:
We can derive the `sequenceNumber` by finding the action with this `rowID`?

Reply:
To clarify, it's in the backend and we could make this change hypothetically if we tell PHP that we want it to load everything from "the most recent to the action we are looking for + 10 more" - surely we can do this somehow?

Reply:
Seems like a better design vs. continuously polling for the next 50 actions until we find the one we are looking for? (also stop me if I'm missing something - seems I am missing quite a lot haha)

Reply:
Ah yes i definitely think there is room here for improvements. But yes they do require some change to the backend so i feel like they should be part of future phase improvements.

> We can derive the sequenceNumber by finding the action with this rowID?

Correct, backend can but not the front end.

Yeah i definitely think a simple future phase improvement would be to,
- if reportActionID is not cached then just reload component with bidirectional scroll pointing to the desired reportActionID. I don't know if its worth loading and waiting for 50,100 or 150 comments to render before scrolling when bidirectional could just go there directly and then work on loading actionIDs below and above it.

I feel like the first phase is just something quick to achieve and polling to load more will be removed with bidirectional scroll. Thats my thought atleast. Let me know if you prefer a different direction.

Reply:
As discussed i feel like the first version will be pretty basic and then once we have bidirectional scroll in plan we'll need to have a design doc or plan of action to discuss how to implement this well in a much more performant way. So i think its okay for now. Feel free to let me know if you disagree.

Reply:
I pretty much agree with Marc here. Having a V1 that only lets us scroll back 150 comments seems a bit useless.